src/Model/FileSupport/BuyRequest/CustomizableOptionDataProvider.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/quote-graphql
+ * @link    https://github.com/scandipwa/quote-graphql
+ */
+
+declare(strict_types=1);
+
+namespace ScandiPWA\QuoteGraphQl\Model\FileSupport\BuyRequest;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Model\Cart\Data\CartItem;
+use Magento\Framework\Filesystem;
+use Magento\Quote\Model\Cart\BuyRequest\BuyRequestDataProviderInterface;
+
+/**
+ * Extract buy request elements require for custom options
+ */
+class CustomizableOptionDataProvider implements BuyRequestDataProviderInterface
+{
+    const OPTION_TYPE = 'custom-option';
+    const QUOTE_MEDIA_PATH = 'custom_options/quote/';
+    const ORDER_MEDIA_PATH = 'custom_options/order/';
+
+    /**
+     * @var string
+     */
+    protected $mediaPath;
+
+    /**
+     * @param Filesystem $filesystem
+     */
+    public function __construct(
+        Filesystem $filesystem
+    ) {
+        $this->mediaPath = $filesystem->getDirectoryRead(DirectoryList::MEDIA)->getAbsolutePath();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @throws LocalizedException
+     */
+    public function execute(CartItem $cartItem): array
+    {
+        $customizableOptionsData = [];
+
+        foreach ($cartItem->getSelectedOptions() as $optionData) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            $optionData = \explode('/', base64_decode($optionData->getId()));
+
+            if ($this->isProviderApplicable($optionData) === false) {
+                continue;
+            }
```

Comment:
If there is any function or assignment going before or after 'if' block add please white space.